### DISCUSSION TITLE
How to get the hidden layer outputs of the Deep GP model

### DISCUSSION THREAD

**Discussion by MartianonEarth** [cornellius-gp/gpytorch#2368] [Q&A]
Hello, the mighty community,

I am new to the GPytorch. I try to run the example of the Deep GP, which works fine. 

I want to get the outputs of hidden layers of the deep GP model prediction, but I can not figure out a way to do it. 
There is no such example in the Documentation.

The only hidden layer outputs I get are for hidden layers in the deep GP model but not for its predictions.

Can you please shed any light on this?

Many thank in advance.

**Comment by gpleiss** [comment]
Following the [Deep GP example](https://docs.gpytorch.ai/en/stable/examples/05_Deep_Gaussian_Processes/Deep_Gaussian_Processes.html), you could run

```python
hidden_layer_dist = self.hidden_layer(x)
```
